kimchi/src/prover.rs

```diff
@@ -786,13 +787,23 @@ where
             // xor
             {
                 if index.cs.xor_selector_poly.is_some() {
-                    let xor = xor::combined_constraints(&all_alphas).evaluations(&env);
+                    let xor = Xor16::combined_constraints(&all_alphas).evaluations(&env);
                     assert_eq!(xor.domain().size, t4.domain().size);
                     t4 += &xor;
                     check_constraint!(index, xor);
                 }
             }
 
+            // rot
+            {
+                if index.cs.rot_selector_poly.is_some() {
+                    let rot = Rot64::combined_constraints(&all_alphas).evaluations(&env);
+                    assert_eq!(rot.domain().size, t4.domain().size);
+                    t4 += &rot;
```

Comment:
This should be  `t8`, and the assertion above is incorrect.

Reply:
Oh thanks, I was not really sure what was the domain being used for each gate... For example chacha uses t4 and range-check uses t8. The one for `xor` is fine though? meaning t4. Is this something that has something to do with the number of rows each type requires?

Reply:
It has to do with the degree of the polynomial. This has degree > 4, so it needs to be evaluated over the `d8` domain instead of the `d4` one.

Reply:
Oh wow... You mean including the selector polynomial multiplication, right? Thanks for this explanation, it will be the most useful!!